docs/tla/TCommit.tla

```diff
@@ -0,0 +1,45 @@
+(* MIT License Copyright (c) 2020 CQFN
+ *    https://github.com/cqfn/degitx/blob/master/LICENSE
+ *)
+------------------------------ MODULE TCommit ------------------------------
+CONSTANT RM
+
+VARIABLES rmState
+
+\* Invariants
+TCTypeOK == rmState \in [RM -> {"working", "prepared", "pre-committed", "committed", "aborted"}]
+
+TCConsistent == \A r1, r2 \in RM : ~ /\ rmState[r1] = "aborted"
+                                     /\ rmState[r2] = "committed"
+
+----------------------------------------------------
+
+TCInit == rmState = [r \in RM |-> "working"]
+
+canPreCommit == \A r \in RM : rmState[r] \in {"prepared", "pre-committed"}
+
+canCommit == \A r \in RM : rmState[r] \in {"pre-committed", "committed"}
+
+canAbort == \A r \in RM : rmState[r] \notin {"pre-committed", "committed"}
+
+Prepare(r) == /\ rmState[r] = "working"
+              /\ rmState' = [rmState EXCEPT ![r] = "prepared"]
```

Comment:
Why only `Prepare` was moved to separate definition? I think it'd be better to either move all state transitions to definitions (like `Prepare(r)`, `PreCommit(r)`, `Commit(r)` and `Abort(r)`) and define `Decide` as any of these transition; or move `Prepare(r)` to `Decide(r)` definition, since `Decide(r)` is not too complex. So both ways could work here.